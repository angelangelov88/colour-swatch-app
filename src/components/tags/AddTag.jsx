import React from 'react'
import classNames from 'classnames'
import PropTypes from 'prop-types'
import { AddRemoveBtnComponent } from './AddRemoveBtnComponent'

const AddTag = (props) => {
  let {
    children,
    className,
    size,
    iconType,
    iconSize
  } = props

  const classes = classNames(
    "font-ibm inline-flex items-center font-medium w-20 rounded-full text-gray-90 hover:cursor-pointer hover:text-primary-light active:bg-primary-10 active:cursor-default",
    size === "sm" && "px-2 py-0.5 text-xs",
    size === "lg" && "py-1 text-sm",
    
    className
  );

  return (
    <div className={classes}>
      <AddRemoveBtnComponent 
        icon={iconType}
        iconSize={iconSize}
      />
      <span>{children}</span>
    </div>
  )
}

AddTag.defaultProps = {
  children: "Add Tag",
  size: "lg",
  iconType: "x",
  iconSize: "lg"
}

AddTag.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  size: PropTypes.oneOf(["sm", "lg"]),
  iconType: PropTypes.oneOf(["plus","x","none"]),
  iconSize: PropTypes.oneOf(["sm", "lg"]),
}

export { AddTag }