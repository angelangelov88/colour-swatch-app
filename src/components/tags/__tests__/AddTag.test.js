import renderer from 'react-test-renderer'
import {AddTag} from "../AddTag"

it('should render the tag with its default settings - children: "Add Tag", size: "lg"', () => {
  const component = renderer.create(
  <AddTag />,
  );
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
});

it('should render the tag with text - This is a tag', () => {
  const component = renderer.create(
  <AddTag 
    children="This is a tag"
  />,
  );
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
});


it('should render a small tag with text Tagged', () => {
  const component = renderer.create(
  <AddTag 
    children="Tagged"
    size="sm"
  />,
  );
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
});


it('should render a large tag with remove icon', () => {
  const component = renderer.create(
  <AddTag 
    size="sm"
  />,
  );
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
});


it('should render a small tag with text - New tag', () => {
  const component = renderer.create(
  <AddTag 
    children="New Tag"
    size="sm"
  />,
  );
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
});