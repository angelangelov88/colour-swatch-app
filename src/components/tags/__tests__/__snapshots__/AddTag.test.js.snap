// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a large tag with remove icon 1`] = `
<div
  className="font-ibm inline-flex items-center w-max pr-2 font-medium rounded-full text-gray-90 border-4 border-transparent group hover:cursor-pointer hover:text-primary-light active:bg-primary-10 active:cursor-default focus:outline focus:outline-4 focus:outline-primary-30 h-5 text-xs"
  tabIndex={0}
>
  <div
    className="font-ibm flex items-center justify-center rounded-xl text-gray-60 group-hover:text-primary-light hover:cursor-pointer w-4 h-4 p-0.5"
  >
    <svg
      aria-hidden="true"
      fill="none"
      stroke="currentColor"
      strokeWidth={2}
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 4v16m8-8H4"
        strokeLinecap="round"
        strokeLinejoin="round"
      />
    </svg>
  </div>
  <span>
    Add Tag
  </span>
</div>
`;

exports[`should render a small tag with text - New tag 1`] = `
<div
  className="font-ibm inline-flex items-center w-max pr-2 font-medium rounded-full text-gray-90 border-4 border-transparent group hover:cursor-pointer hover:text-primary-light active:bg-primary-10 active:cursor-default focus:outline focus:outline-4 focus:outline-primary-30 h-5 text-xs"
  tabIndex={0}
>
  <div
    className="font-ibm flex items-center justify-center rounded-xl text-gray-60 group-hover:text-primary-light hover:cursor-pointer w-4 h-4 p-0.5"
  >
    <svg
      aria-hidden="true"
      fill="none"
      stroke="currentColor"
      strokeWidth={2}
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 4v16m8-8H4"
        strokeLinecap="round"
        strokeLinejoin="round"
      />
    </svg>
  </div>
  <span>
    New Tag
  </span>
</div>
`;

exports[`should render a small tag with text Tagged 1`] = `
<div
  className="font-ibm inline-flex items-center w-max pr-2 font-medium rounded-full text-gray-90 border-4 border-transparent group hover:cursor-pointer hover:text-primary-light active:bg-primary-10 active:cursor-default focus:outline focus:outline-4 focus:outline-primary-30 h-5 text-xs"
  tabIndex={0}
>
  <div
    className="font-ibm flex items-center justify-center rounded-xl text-gray-60 group-hover:text-primary-light hover:cursor-pointer w-4 h-4 p-0.5"
  >
    <svg
      aria-hidden="true"
      fill="none"
      stroke="currentColor"
      strokeWidth={2}
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 4v16m8-8H4"
        strokeLinecap="round"
        strokeLinejoin="round"
      />
    </svg>
  </div>
  <span>
    Tagged
  </span>
</div>
`;

exports[`should render the tag with its default settings - children: "Add Tag", size: "lg" 1`] = `
<div
  className="font-ibm inline-flex items-center w-max pr-2 font-medium rounded-full text-gray-90 border-4 border-transparent group hover:cursor-pointer hover:text-primary-light active:bg-primary-10 active:cursor-default focus:outline focus:outline-4 focus:outline-primary-30 h-6 text-sm"
  tabIndex={0}
>
  <div
    className="font-ibm flex items-center justify-center rounded-xl text-gray-60 group-hover:text-primary-light hover:cursor-pointer w-5 h-5 p-1"
  >
    <svg
      aria-hidden="true"
      fill="none"
      stroke="currentColor"
      strokeWidth={2}
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 4v16m8-8H4"
        strokeLinecap="round"
        strokeLinejoin="round"
      />
    </svg>
  </div>
  <span>
    Add Tag
  </span>
</div>
`;

exports[`should render the tag with text - This is a tag 1`] = `
<div
  className="font-ibm inline-flex items-center w-max pr-2 font-medium rounded-full text-gray-90 border-4 border-transparent group hover:cursor-pointer hover:text-primary-light active:bg-primary-10 active:cursor-default focus:outline focus:outline-4 focus:outline-primary-30 h-6 text-sm"
  tabIndex={0}
>
  <div
    className="font-ibm flex items-center justify-center rounded-xl text-gray-60 group-hover:text-primary-light hover:cursor-pointer w-5 h-5 p-1"
  >
    <svg
      aria-hidden="true"
      fill="none"
      stroke="currentColor"
      strokeWidth={2}
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 4v16m8-8H4"
        strokeLinecap="round"
        strokeLinejoin="round"
      />
    </svg>
  </div>
  <span>
    This is a tag
  </span>
</div>
`;
